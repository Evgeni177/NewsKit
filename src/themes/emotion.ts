import React from 'react';
import {withTheme as _withTheme, useTheme as _useTheme} from 'emotion-theming';
import {Theme} from './creator';

const withTheme = (_withTheme as unknown) as <P>(
  comp: React.ComponentType<P & {theme: Theme}>,
) => React.ComponentType<P & {theme?: Theme}>;

const useTheme = (_useTheme as unknown) as () => Theme;

export {withTheme, useTheme};
export {ThemeProvider} from 'emotion-theming';
export {Global} from '@emotion/core';
