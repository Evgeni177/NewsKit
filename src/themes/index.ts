export {Borders, BorderKeys} from './mappers/borders';
export {ColorKeys, Colors} from './mappers/colors';
export {TypePresetKeys, TypePresets} from './mappers/type-presets';

export {colorPrimitives, ColorPrimitives} from './newskit-light/colors';
export {
  AnimationEasing as Animation,
  AnimationEasingKeys as AnimationKeys,
  AnimationDuration,
  AnimationDurationKeys,
} from './newskit-light/animation';
export {BorderRadius, BorderRadiusShape} from './newskit-light/border-radius';
export {Breakpoints, BreakpointKeys} from './newskit-light/breakpoints';
export {Sizing, SizingKeys} from './newskit-light/sizing';
export {FontPrimitives, FontPrimitivesKeys} from './newskit-light/fonts';

export {newskitLightTheme} from './newskit-light';
export {theSunTheme} from './the-sun';
export {theTimesTheme} from './the-times';

export * from './emotion';
export * from './creator';
