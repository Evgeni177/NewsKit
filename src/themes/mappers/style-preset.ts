import {
  BackgroundRepeatProperty,
  BackgroundPositionProperty,
  BorderStyleProperty,
  BoxShadowProperty,
  BackgroundColorProperty,
  BackgroundImageProperty,
  BorderColorProperty,
  ColorProperty,
} from 'csstype';
import {BorderPrimitives} from '../newskit-light/borders';
import {ColorPrimitives} from '../newskit-light/colors';
import {BorderRadiusShape, BorderRadius} from '../newskit-light/border-radius';
import {Sizing} from '../newskit-light/spacing';

export interface StylePresetStyles {
  boxShadow?: BoxShadowProperty; // shadow
  backgroundColor?: BackgroundColorProperty;
  backgroundImage?: BackgroundImageProperty;
  backgroundRepeat?: BackgroundRepeatProperty;
  backgroundPosition?: BackgroundPositionProperty<string>;
  borderStyles?: BorderStyleProperty;
  borderColor?: BorderColorProperty;
  borderWidth?: string;
  color?: ColorProperty;
  borderRadius?: Sizing; // shape
  iconColor?: ColorProperty;
}
export type StylePresetStyleKeys = keyof StylePresetStyles;

export interface StylePresetStates {
  // component states
  current?: StylePresetStyles;
  loading?: StylePresetStyles;
  // baseline css
  base?: StylePresetStyles;
  // css pseudo classes
  hover?: StylePresetStyles;
  focus?: StylePresetStyles;
  active?: StylePresetStyles;
  visited?: StylePresetStyles;
  disabled?: StylePresetStyles;
}
export type StylePresetStateKeys = keyof StylePresetStates;

interface CreateStylePreset {
  colorPrimitives: ColorPrimitives;
  borderPrimitives: BorderPrimitives;
  borderRadiusPrimitives: BorderRadius;
}

export const createStylePresets = ({
  colorPrimitives,
  borderPrimitives,
  borderRadiusPrimitives,
}: CreateStylePreset) => ({
  interactive010: {
    base: {
      backgroundColor: colorPrimitives.brand010,
      color: colorPrimitives.inkInverse,
      borderRadius: borderRadiusPrimitives[BorderRadiusShape.Rounded],
      iconColor: colorPrimitives.inkInverse,
    },
    hover: {
      backgroundColor: colorPrimitives.blue070,
    },
    active: {
      backgroundColor: colorPrimitives.blue080,
    },
    focus: {
      borderWidth: borderPrimitives.borderWidth020,
    },
    disabled: {
      backgroundColor: colorPrimitives.neutral020,
      color: colorPrimitives.inkNonEssential,
    },
  } as StylePresetStates,
  interactive010Inverse: {
    base: {
      color: colorPrimitives.inkBrand,
      borderRadius: borderRadiusPrimitives[BorderRadiusShape.Rounded],
      iconColor: colorPrimitives.inkBrand,
    },
    hover: {
      backgroundColor: colorPrimitives.neutral010,
      iconColor: colorPrimitives.blue070,
    },
    active: {
      backgroundColor: colorPrimitives.neutral010,
      iconColor: colorPrimitives.blue070,
    },
    focus: {
      borderWidth: borderPrimitives.borderWidth020,
    },
    disabled: {
      backgroundColor: colorPrimitives.neutral020,
      color: colorPrimitives.inkNonEssential,
    },
  } as StylePresetStates,
  interactive020: {
    base: {
      backgroundColor: colorPrimitives.white,
      borderStyles: 'solid',
      borderColor: colorPrimitives.brand010,
      borderWidth: borderPrimitives.borderWidth010,
      color: colorPrimitives.inkBrand,
      borderRadius: borderRadiusPrimitives[BorderRadiusShape.Rounded],
      iconColor: colorPrimitives.inkBrand,
    },
    hover: {
      boxShadow: '0 0 0 4px rgba(200,228,253,0.60)',
    },
    active: {
      boxShadow: '0 0 0 4px rgba(200,228,253,0.60)',
    },
    disabled: {
      backgroundColor: colorPrimitives.neutral020,
    },
  } as StylePresetStates,
  interactive030: {
    base: {
      backgroundColor: colorPrimitives.neutral040,
      borderRadius: borderRadiusPrimitives[BorderRadiusShape.Rounded],
    },
    disabled: {
      backgroundColor: colorPrimitives.neutral020,
    },
  } as StylePresetStates,
  interactive040: {
    base: {
      backgroundColor: colorPrimitives.neutral030,
      borderRadius: borderRadiusPrimitives[BorderRadiusShape.Rounded],
    },
    disabled: {
      backgroundColor: colorPrimitives.neutral020,
    },
  } as StylePresetStates,
  maskPointed010: {
    base: {
      borderRadius: borderRadiusPrimitives[BorderRadiusShape.Squared],
    },
    loading: {
      backgroundColor: colorPrimitives.skeletonLight,
      borderRadius: borderRadiusPrimitives[BorderRadiusShape.Squared],
      iconColor: colorPrimitives.inkSubtle,
    },
  } as StylePresetStates,
  maskSemiRounded: {
    base: {
      borderRadius: borderRadiusPrimitives[BorderRadiusShape.SemiRounded],
    },
    loading: {
      backgroundColor: colorPrimitives.skeletonLight,
      borderRadius: borderRadiusPrimitives[BorderRadiusShape.SemiRounded],
      iconColor: colorPrimitives.inkSubtle,
    },
  } as StylePresetStates,
  maskRound: {
    base: {
      borderRadius: borderRadiusPrimitives[BorderRadiusShape.Rounded],
    },
    loading: {
      backgroundColor: colorPrimitives.skeletonLight,
      borderRadius: borderRadiusPrimitives[BorderRadiusShape.Rounded],
      iconColor: colorPrimitives.inkSubtle,
    },
  } as StylePresetStates,
});

export type StylePresets = ReturnType<typeof createStylePresets> &
  Record<string, StylePresetStates>;
