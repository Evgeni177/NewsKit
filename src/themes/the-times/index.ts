import {createTheme, Theme} from '../creator';

export const theTimesTheme = createTheme('the-times-theme', {
  colorOverrides: {
    inkContrast: '#1D1D1B',
    inkBase: '#333333',
    inkSubtle: '#696969',
    inkNonEssential: '#4D4D4D',

    interface010: '#F5F5F5',
    interface020: '#FFFFFF',
    interface030: '#F0F0F0',
    interface040: '#DBDBDB',
    interface050: '#272D34',

    semanticNegative010: '#C51D24',
  },
  fontOverrides: {
    fontFamily1: 'TimesModern-Regular',
    fontFamily2: 'TimesDigitalW04-Regular',
    fontFamily3: 'GillSansMTStd-Medium',
  },
  themeOverrider: ({colors, borders}) => ({
    borders: {
      tagBorderWidth: borders.borderWidth010,
    } as Theme['borders'],
    colors: {
      tagBorder: colors.interface040,
      tagHoverBorder: colors.interface040,

      linkText: colors.inkLink,
      linkActive: colors.inkLink,
      linkHover: colors.inkLink,
      linkVisited: colors.inkLink,

      standfirstText: colors.inkSubtle,
    } as Theme['colors'],
  }),
});
