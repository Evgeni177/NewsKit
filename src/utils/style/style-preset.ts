import {
  Theme,
  StylePresetStates,
  StylePresetStyleKeys,
  StylePresetStyles,
  StylePreset,
} from '../../theme';
import {filterObject, rejectObject} from '../filter-object';
import {ThemeProp} from '../style-types';
import {getDefaultedValue, getResponsiveValueFromTheme} from './base';
import {CSSObject} from './emotion';
import {MQ} from './types';

export interface GetStylePresetFromThemeOptions {
  nestedCssSelector?: string;
  isLoading?: boolean;
  isSelected?: boolean;
  isDisabled?: boolean;
  isInvalid?: boolean;
  isValid?: boolean;
  isSvg?: boolean;
  omitStates?: StylePresetStates[];
  filterStates?: StylePresetStates[];
  omitStyles?: StylePresetStyleKeys[];
  filterStyles?: StylePresetStyleKeys[];
}

/* When we are not on directly on a svg we need to add an 
  additional css selector to increase the specificity, allowing
  us to overrule the default color. Icon button is an example 
  of this in action */
const getCssSvgFillObject = (iconColor: string, isSvg: boolean): object =>
  isSvg
    ? {fill: iconColor}
    : {
        svg: {
          fill: iconColor,
        },
      };

export const getPresetStyles = (
  presetStyles: StylePresetStyles,
  options?: GetStylePresetFromThemeOptions,
) => {
  const {filterStyles = null, omitStyles = [], isSvg: isSvg = false} =
    options || {};
  const {iconColor, placeholderColor, ...cssObject} = filterStyles
    ? filterObject(presetStyles, filterStyles)
    : rejectObject(presetStyles, omitStyles);
  if (iconColor) {
    return {
      ...cssObject,
      ...getCssSvgFillObject(iconColor, isSvg),
    } as CSSObject;
  }

  if (placeholderColor) {
    return {
      ...cssObject,
      '::placeholder': {
        color: placeholderColor,
      },
    } as CSSObject;
  }

  return cssObject as CSSObject;
};

const getPresetStates = (
  stylePreset: StylePreset,
  options?: GetStylePresetFromThemeOptions,
) => {
  const {
    omitStates = [],
    filterStates = [],
    isSelected = false,
    isLoading = false,
    isDisabled = false,
    isInvalid = false,
    isValid = false,
  } = options || {};
  const {selected, loading, invalid, valid, ...presetStates} =
    filterStates && filterStates.length
      ? filterObject(stylePreset, filterStates)
      : rejectObject(stylePreset, omitStates);
  const stateOverrides =
    (isDisabled && presetStates.disabled) ||
    (isLoading && loading) ||
    (isSelected && selected) ||
    (isInvalid && invalid) ||
    (isValid && valid) ||
    undefined;
  if (stateOverrides) {
    const {base = {}} = presetStates;
    return {base: {...base, ...stateOverrides}};
  }

  return presetStates;
};

const getStylePresetValueFromTheme = (
  stylePreset: StylePreset,
  options?: GetStylePresetFromThemeOptions,
) =>
  Object.entries(getPresetStates(stylePreset, options)).reduce(
    (acc, [stateKey, presetState]) => {
      if (presetState) {
        const nestedCssSelector = options && options.nestedCssSelector;
        const selector =
          stateKey === 'disabled'
            ? `:${stateKey} ${nestedCssSelector || ''}`
            : `:${stateKey}:not(:disabled) ${nestedCssSelector || ''}`;
        const styles = getPresetStyles(presetState, options);
        if (stateKey === 'base') {
          if (nestedCssSelector) {
            acc[nestedCssSelector] = styles;
            return acc;
          }
          return {...acc, ...styles};
        }
        acc[selector] = styles;
      }

      return acc;
    },
    {} as CSSObject,
  );

export const getStylePresetFromTheme = <Props extends ThemeProp>(
  defaultToken?: MQ<string>,
  customProp?: Exclude<keyof Props, 'theme'>,
  options?: GetStylePresetFromThemeOptions,
) => (props: Props) => {
  const stylePreset = getResponsiveValueFromTheme('stylePresets')(
    defaultToken,
    customProp,
  )(props) as Partial<StylePreset> | Array<[string, StylePreset]>;

  if (Array.isArray(stylePreset)) {
    return stylePreset.reduce((acc, [mq, preset]) => {
      acc[mq] = getStylePresetValueFromTheme(preset, options);
      return acc;
    }, {} as CSSObject);
  }
  return stylePreset ? getStylePresetValueFromTheme(stylePreset, options) : '';
};

export const getStylePreset = getDefaultedValue(
  getStylePresetFromTheme,
  'stylePreset',
);

export const getSingleStylePreset = (
  {stylePresets}: Theme,
  state: StylePresetStates,
  cssProp: Exclude<StylePresetStyleKeys, 'borderRadius'>,
  defaultToken: string,
  customToken?: string,
) => {
  const preset =
    (customToken && stylePresets[customToken]) || stylePresets[defaultToken];
  return (preset[state] || {})[cssProp] || '';
};
