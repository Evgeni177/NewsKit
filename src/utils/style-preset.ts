import {ThemeProp, css, CSSObject} from './style';
import {
  StylePresetStyles,
  StylePresetStyleKeys,
  StylePresetStateKeys,
  StylePresetStates,
} from '../themes/mappers/style-preset';
import {filterObject} from './filter-object';
import {SizingKeys} from '../themes/newskit-light/spacing';

export interface GetStylePresetFromThemeOptions {
  isLoading?: boolean;
  isCurrent?: boolean;
  isDisabled?: boolean;
  omitStates?: StylePresetStateKeys[];
  omitStyles?: StylePresetStyleKeys[];
  borderRadiusSize?: SizingKeys;
}

type GetPresetStylesOptions = Pick<
  GetStylePresetFromThemeOptions,
  'omitStyles' | 'borderRadiusSize'
>;

const getPresetStyles = (
  presetStyles: StylePresetStyles,
  options?: GetPresetStylesOptions,
) => {
  const {omitStyles = [], borderRadiusSize = undefined} = options || {};

  const {borderRadius: borderRadiusSizing, ...styles} = filterObject(
    presetStyles,
    omitStyles,
  );

  const borderRadius =
    borderRadiusSizing && borderRadiusSize
      ? borderRadiusSizing[borderRadiusSize]
      : '';

  return {
    ...styles,
    borderRadius,
  };
};

const cssTmpl = (
  {iconColor = '', ...cssObject}: ReturnType<typeof getPresetStyles>,
  selector: string = '',
) => css`
  ${selector && `:${selector} {`}
  ${cssObject as CSSObject}
  ${iconColor &&
    `
    svg {
      fill: ${iconColor};
      color: ${iconColor};
    }
  `}
  ${selector && '}'}
`;

export const getStylePresetFromTheme = <Props extends ThemeProp>(
  defaultToken?: string,
  customProp?: Exclude<keyof Props, 'theme'>,
  options?: GetStylePresetFromThemeOptions,
) => ({theme, ...props}: Props) => {
  const {
    omitStates = [],
    isCurrent = false,
    isLoading = false,
    isDisabled = false,
    ...presetOptions
  } = options || {};

  const stylePreset =
    (customProp &&
      theme.stylePresets[(props[customProp] as unknown) as string]) ||
    (defaultToken && theme.stylePresets[defaultToken]);

  if (!stylePreset) {
    return '';
  }

  const {current, loading, ...presetStates} = filterObject(
    stylePreset as StylePresetStates,
    omitStates,
  );

  const stateOverrides =
    (isDisabled && presetStates.disabled) ||
    (isLoading && loading) ||
    (isCurrent && current) ||
    undefined;

  if (stateOverrides) {
    const {base = {}} = presetStates;
    return cssTmpl(
      getPresetStyles({...base, ...stateOverrides}, presetOptions),
    );
  }

  return Object.entries(presetStates).map(([stateKey, presetState]) => {
    if (presetState) {
      const presetStyle = getPresetStyles(presetState, presetOptions);

      if (stateKey === 'base') {
        return cssTmpl(presetStyle);
      }

      const selector =
        stateKey === 'disabled' ? stateKey : `${stateKey}:not(:disabled)`;

      return cssTmpl(presetStyle, selector);
    }

    return '';
  });
};
