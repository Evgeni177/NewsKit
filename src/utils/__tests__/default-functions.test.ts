import {
  styled,
  getDefaultTypePreset,
  getDefaultMarginPreset,
  getDefaultPaddingPreset,
} from '../style';
import {renderToFragmentWithTheme} from '../../test/test-utils';
import {getDefaultStylePreset} from '../style-preset';
import {createTheme} from '../../themes';

describe('get default presets functions', () => {
  const theme = createTheme('test-theme', {
    themeOverrider: () => ({
      defaultPresets: {
        articleHeadline: {
          heading: {
            stylePreset: 'articleHeadlineContent',
            typePreset: {
              xs: 'headline100',
              md: 'headline200',
            },
          },
          kicker: {
            typePreset: 'headline100',
            margin: {
              xs: 'spaceInline030',
              sm: 'spaceInline030',
              md: 'spaceInline040',
              lg: 'spaceInline040',
            },
            padding: {
              xs: 'spaceStack020',
              sm: 'spaceStack030',
              md: 'spaceStack040',
            },
          },
        },
      },
    }),
  });

  test('getDefaultTypePreset handles object breakpoints', () => {
    const defaultTestSurface = styled.div`
      display: block;
      ${getDefaultTypePreset('articleHeadline.heading')}
    `;

    const fragment = renderToFragmentWithTheme(
      defaultTestSurface,
      undefined,
      theme,
    );
    expect(fragment).toMatchSnapshot();
  });

  test('getDefaultTypePreset with breakpoints and crop', () => {
    const defaultTestSurface = styled.div`
      display: block;

      ${getDefaultTypePreset('articleHeadline.heading', undefined as never, {
        withCrop: true,
      })}
    `;

    const fragment = renderToFragmentWithTheme(
      defaultTestSurface,
      undefined,
      theme,
    );
    expect(fragment).toMatchSnapshot();
  });

  test('getDefaultTypePreset without breakpoints and with crop', () => {
    const defaultTestSurface = styled.div`
      display: block;
      ${getDefaultTypePreset('articleHeadline.kicker', undefined as never, {
        withCrop: true,
      })}
    `;

    const fragment = renderToFragmentWithTheme(
      defaultTestSurface,
      undefined,
      theme,
    );
    expect(fragment).toMatchSnapshot();
  });

  test('getDefaultTypePreset with customProp', () => {
    const defaultTestSurface = styled.div<{customProp: string}>`
      display: block;
      ${getDefaultTypePreset('articleHeadline.heading', 'customProp')}
    `;

    const fragment = renderToFragmentWithTheme(
      defaultTestSurface,
      {
        customProp: 'heading010',
      },
      theme,
    );
    expect(fragment).toMatchSnapshot();
  });

  test('getDefaultStylePreset handles string preset', () => {
    const defaultTestSurface = styled.div`
      display: block;
      ${getDefaultStylePreset('articleHeadline.heading')}
    `;
    const fragment = renderToFragmentWithTheme(
      defaultTestSurface,
      undefined,
      theme,
    );
    expect(fragment).toMatchSnapshot();
  });

  test('getDefaultStylePreset overwrite preset', () => {
    const defaultTestSurface = styled.p<{customProp: string}>`
      display: block;
      ${getDefaultStylePreset('articleHeadline.heading', 'customProp')}
    `;

    const fragment = renderToFragmentWithTheme(
      defaultTestSurface,
      {
        customProp: 'flagSolidLive',
      },
      theme,
    );
    expect(fragment).toMatchSnapshot();
  });

  test('getDefaultMarginPreset handle breakpoints preset', () => {
    const defaultTestSurface = styled.p`
      display: block;
      ${getDefaultMarginPreset('articleHeadline.kicker')}
    `;

    const fragment = renderToFragmentWithTheme(
      defaultTestSurface,
      undefined,
      theme,
    );
    expect(fragment).toMatchSnapshot();
  });

  test('getDefaultMarginPreset handle overwrite variable', () => {
    const defaultTestSurface = styled.p<{customProp: string}>`
      display: block;
      ${getDefaultMarginPreset('articleHeadline.kicker', 'customProp')}
    `;

    const fragment = renderToFragmentWithTheme(
      defaultTestSurface,
      {
        customProp: 'spaceStack070',
      },
      theme,
    );
    expect(fragment).toMatchSnapshot();
  });

  test('getDefaultPaddingPreset handle breakpoints preset', () => {
    const defaultTestSurface = styled.p`
      display: block;
      ${getDefaultPaddingPreset('articleHeadline.kicker')}
    `;

    const fragment = renderToFragmentWithTheme(
      defaultTestSurface,
      undefined,
      theme,
    );
    expect(fragment).toMatchSnapshot();
  });

  test('getDefaultPaddingPreset handle overwrite variable', () => {
    const defaultTestSurface = styled.p<{customProp: string}>`
      display: block;
      ${getDefaultPaddingPreset('articleHeadline.kicker', 'customProp')}
    `;

    const fragment = renderToFragmentWithTheme(
      defaultTestSurface,
      {
        customProp: 'spaceInset050',
      },
      theme,
    );
    expect(fragment).toMatchSnapshot();
  });
});
