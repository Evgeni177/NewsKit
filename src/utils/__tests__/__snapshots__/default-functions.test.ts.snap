// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get default presets functions getMarginPreset for responsive tokens when props.overrides is set with breakpoint object with no override path provided: breakpoint-override-object:no-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "margin": "theOverrideToken-value-space",
  },
  "margin": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getMarginPreset for responsive tokens when props.overrides is set with breakpoint object with no override path provided: breakpoint-override-object:with-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "margin": "theOverrideToken-value-space",
  },
  "margin": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getMarginPreset for responsive tokens when props.overrides is set with breakpoint object with override path as empty string: breakpoint-override-object:no-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "margin": "theOverrideToken-value-space",
  },
  "margin": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getMarginPreset for responsive tokens when props.overrides is set with breakpoint object with override path as empty string: breakpoint-override-object:with-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "margin": "theOverrideToken-value-space",
  },
  "margin": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getMarginPreset for responsive tokens when props.overrides is set with breakpoint object with override path as empty string: breakpoint-override-object:with-path-end 2`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "margin": "theOverrideToken-value-space",
  },
  "margin": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getMarginPreset for responsive tokens when props.overrides is set with breakpoint object with override path: breakpoint-override-object:with-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "margin": "theOverrideToken-value-space",
  },
  "margin": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getMarginPreset for responsive tokens with no override path provided, with and without token type at the end: default:no-path-end 1`] = `
Object {
  "@media screen and (min-width: 960px)": Object {
    "margin": "theDefaultToken-value-space",
  },
  "margin": "theDefaultToken-value-space",
}
`;

exports[`get default presets functions getMarginPreset for responsive tokens with no override path provided, with and without token type at the end: default:with-path-end 1`] = `
Object {
  "@media screen and (min-width: 960px)": Object {
    "margin": "theDefaultToken-value-space",
  },
  "margin": "theDefaultToken-value-space",
}
`;

exports[`get default presets functions getPaddingPreset for responsive tokens when props.overrides is set with breakpoint object with no override path provided: breakpoint-override-object:no-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "padding": "theOverrideToken-value-space",
  },
  "padding": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getPaddingPreset for responsive tokens when props.overrides is set with breakpoint object with no override path provided: breakpoint-override-object:with-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "padding": "theOverrideToken-value-space",
  },
  "padding": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getPaddingPreset for responsive tokens when props.overrides is set with breakpoint object with override path as empty string: breakpoint-override-object:no-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "padding": "theOverrideToken-value-space",
  },
  "padding": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getPaddingPreset for responsive tokens when props.overrides is set with breakpoint object with override path as empty string: breakpoint-override-object:with-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "padding": "theOverrideToken-value-space",
  },
  "padding": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getPaddingPreset for responsive tokens when props.overrides is set with breakpoint object with override path as empty string: breakpoint-override-object:with-path-end 2`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "padding": "theOverrideToken-value-space",
  },
  "padding": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getPaddingPreset for responsive tokens when props.overrides is set with breakpoint object with override path: breakpoint-override-object:with-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "padding": "theOverrideToken-value-space",
  },
  "padding": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getPaddingPreset for responsive tokens with no override path provided, with and without token type at the end: default:no-path-end 1`] = `
Object {
  "@media screen and (min-width: 960px)": Object {
    "padding": "theDefaultToken-value-space",
  },
  "padding": "theDefaultToken-value-space",
}
`;

exports[`get default presets functions getPaddingPreset for responsive tokens with no override path provided, with and without token type at the end: default:with-path-end 1`] = `
Object {
  "@media screen and (min-width: 960px)": Object {
    "padding": "theDefaultToken-value-space",
  },
  "padding": "theDefaultToken-value-space",
}
`;

exports[`get default presets functions getSpacingInline for responsive tokens when props.overrides is set with breakpoint object with no override path provided: breakpoint-override-object:no-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "marginRight": "theOverrideToken-value-space",
  },
  "marginRight": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getSpacingInline for responsive tokens when props.overrides is set with breakpoint object with no override path provided: breakpoint-override-object:with-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "marginRight": "theOverrideToken-value-space",
  },
  "marginRight": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getSpacingInline for responsive tokens when props.overrides is set with breakpoint object with override path as empty string: breakpoint-override-object:no-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "marginRight": "theOverrideToken-value-space",
  },
  "marginRight": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getSpacingInline for responsive tokens when props.overrides is set with breakpoint object with override path as empty string: breakpoint-override-object:with-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "marginRight": "theOverrideToken-value-space",
  },
  "marginRight": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getSpacingInline for responsive tokens when props.overrides is set with breakpoint object with override path as empty string: breakpoint-override-object:with-path-end 2`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "marginRight": "theOverrideToken-value-space",
  },
  "marginRight": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getSpacingInline for responsive tokens when props.overrides is set with breakpoint object with override path: breakpoint-override-object:with-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "marginRight": "theOverrideToken-value-space",
  },
  "marginRight": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getSpacingInline for responsive tokens with no override path provided, with and without token type at the end: default:no-path-end 1`] = `
Object {
  "marginRight": "",
}
`;

exports[`get default presets functions getSpacingInline for responsive tokens with no override path provided, with and without token type at the end: default:with-path-end 1`] = `
Object {
  "marginRight": "",
}
`;

exports[`get default presets functions getSpacingStack for responsive tokens when props.overrides is set with breakpoint object with no override path provided: breakpoint-override-object:no-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "marginBottom": "theOverrideToken-value-space",
  },
  "marginBottom": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getSpacingStack for responsive tokens when props.overrides is set with breakpoint object with no override path provided: breakpoint-override-object:with-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "marginBottom": "theOverrideToken-value-space",
  },
  "marginBottom": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getSpacingStack for responsive tokens when props.overrides is set with breakpoint object with override path as empty string: breakpoint-override-object:no-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "marginBottom": "theOverrideToken-value-space",
  },
  "marginBottom": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getSpacingStack for responsive tokens when props.overrides is set with breakpoint object with override path as empty string: breakpoint-override-object:with-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "marginBottom": "theOverrideToken-value-space",
  },
  "marginBottom": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getSpacingStack for responsive tokens when props.overrides is set with breakpoint object with override path as empty string: breakpoint-override-object:with-path-end 2`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "marginBottom": "theOverrideToken-value-space",
  },
  "marginBottom": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getSpacingStack for responsive tokens when props.overrides is set with breakpoint object with override path: breakpoint-override-object:with-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "marginBottom": "theOverrideToken-value-space",
  },
  "marginBottom": "theOverrideToken-value-space",
}
`;

exports[`get default presets functions getSpacingStack for responsive tokens with no override path provided, with and without token type at the end: default:no-path-end 1`] = `
Object {
  "marginBottom": "",
}
`;

exports[`get default presets functions getSpacingStack for responsive tokens with no override path provided, with and without token type at the end: default:with-path-end 1`] = `
Object {
  "marginBottom": "",
}
`;

exports[`get default presets functions getTypographyPreset for responsive tokens when props.overrides is set with breakpoint object with no override path provided: breakpoint-override-object:no-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "fontFamily": "theOverrideToken-value-typographyPresets",
  },
  "fontFamily": "theOverrideToken-value-typographyPresets",
}
`;

exports[`get default presets functions getTypographyPreset for responsive tokens when props.overrides is set with breakpoint object with no override path provided: breakpoint-override-object:with-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "fontFamily": "theOverrideToken-value-typographyPresets",
  },
  "fontFamily": "theOverrideToken-value-typographyPresets",
}
`;

exports[`get default presets functions getTypographyPreset for responsive tokens when props.overrides is set with breakpoint object with override path as empty string: breakpoint-override-object:no-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "fontFamily": "theOverrideToken-value-typographyPresets",
  },
  "fontFamily": "theOverrideToken-value-typographyPresets",
}
`;

exports[`get default presets functions getTypographyPreset for responsive tokens when props.overrides is set with breakpoint object with override path as empty string: breakpoint-override-object:with-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "fontFamily": "theOverrideToken-value-typographyPresets",
  },
  "fontFamily": "theOverrideToken-value-typographyPresets",
}
`;

exports[`get default presets functions getTypographyPreset for responsive tokens when props.overrides is set with breakpoint object with override path as empty string: breakpoint-override-object:with-path-end 2`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "fontFamily": "theOverrideToken-value-typographyPresets",
  },
  "fontFamily": "theOverrideToken-value-typographyPresets",
}
`;

exports[`get default presets functions getTypographyPreset for responsive tokens when props.overrides is set with breakpoint object with override path: breakpoint-override-object:with-path-end 1`] = `
Object {
  "@media screen and (min-width: 1440px)": Object {
    "fontFamily": "theOverrideToken-value-typographyPresets",
  },
  "fontFamily": "theOverrideToken-value-typographyPresets",
}
`;

exports[`get default presets functions getTypographyPreset for responsive tokens with no override path provided, with and without token type at the end: default:no-path-end 1`] = `
Object {
  "@media screen and (min-width: 960px)": Object {
    "fontFamily": "theDefaultToken-value-typographyPresets",
  },
  "fontFamily": "theDefaultToken-value-typographyPresets",
}
`;

exports[`get default presets functions getTypographyPreset for responsive tokens with no override path provided, with and without token type at the end: default:with-path-end 1`] = `
Object {
  "@media screen and (min-width: 960px)": Object {
    "fontFamily": "theDefaultToken-value-typographyPresets",
  },
  "fontFamily": "theDefaultToken-value-typographyPresets",
}
`;
